### ISSUE TITLE
create-remix templates don't work if a local folder of the same name as the template exists

### ISSUE THREAD

**Issue by IanWitham** [remix-run/remix#2491]
### What version of Remix are you using?

1.3.3

### Steps to Reproduce

> npx create-remix@latest
> Just the basics
> Remix Application Server
> TS or JS
> Run npm install or don't

### Expected Behavior

A new Remix app is created.

### Actual Behavior

An empty package.json and package-lock.json are produced along with a "remix" folder containing the entirety of the remix-run/remix repo.

**Comment by brophdawg11** [comment]
👋 What platform did you see this on?  I'm unable to reproduce on OSX locally:

<img width="858" alt="Screen Shot 2022-03-25 at 9 25 28 AM" src="https://user-images.githubusercontent.com/1609022/160130412-4fe23529-af78-411d-a0e5-d0df2dce6cb2.png">

**Comment by cloudy9101** [comment]
Got the same issue.
`npx create-remix@latest` will download the mono repo remix-run/remix to the newly created project dir.

I tried to install create-remix globally and run it manually, that's works for me. It will **only download the remix-ts template** to the newly created project dir.
```
npm i -g create-remix
create-remix
```

**Comment by cloudy9101** [comment]
Got the same issue.
This will download the mono repo remix-run/remix to remix-jokes
```
npx create-remix@latest
? Where would you like to create your app? ./remix-jokes
```

But if the path has `../` in the beginning, it will **only download the template** to remix-jokes ( of course it's located in the parent dir )
```
npx create-remix@latest
? Where would you like to create your app? ../remix-jokes
```

**Comment by DavOnGit** [comment]
Got the same issue.
I'm on windows 10.

**Comment by styxlab** [comment]
Workaround: `px create-remix@latest  --template remix-ts`

**Comment by IanWitham** [comment]
I'm seeing the bug on macOS Catalina 10.15.7. the `../my-remix-app` workaround works.

I tried running `npm uninstall -g create-remix && npm cache clean --force` first and the problem persisted.

On Windows 10 I _don't_ see the bug.

**Comment by cloudy9101** [comment]
I think the root cause it's that **there's a folder called `remix`** when we run create-remix@latest.
The template name of `Just the basics` is `remix`, the detectTemplateType function will return `local` since `fse.existsSync(template)` will return true if there's a folder called `remix`.

![image](https://user-images.githubusercontent.com/5436425/160223158-8112561f-ed3c-4650-8e67-4ac716bfe260.png)

Then it will run the local logic which copies the `remix` folder to the projectDir.

![image](https://user-images.githubusercontent.com/5436425/160223217-34b864b1-748b-4086-9056-f2e444a7a1da.png)

So the real workaround should be to move your remix folder to another dir or rename it until the bug got fixed.

**Comment by IanWitham** [comment]
That's it! Sir, you're a genius.

**Comment by styxlab** [comment]
@cloudy9101 Thanks for spotting the route cause. However, the workaround I proposed allows you to keep your existing remix folder, that's why I find it more practical.

**Comment by cloudy9101** [comment]
@styxlab Yes, your workaround is great.
Maybe I shouldn't say that rename the remix folder is the real workaround, it's just another way to make it work.

**Comment by MichaelDeBoey** [comment]
I believe this was already fixed by @chaance or @mcansh, but can't find the PR/commit.

Can't reproduce it either.

**Comment by mcansh** [comment]
> I believe this was already fixed by @chaance or @mcansh, but can't find the PR/commit.
> 
> Can't reproduce it either.

@chaance is doing some work in that regard in #2500

the issue here is due to having the remix repo checked out as a sibling directory when using `npx create-remix`

**Comment by kiliman** [comment]
BTW: It doesn't have to be the `remix` **repo**. Just a local folder named `remix`.

I was wondering why every time I tried the RAS template, it was copying all the projects in my `remix` folder.

**Comment by MichaelDeBoey** [comment]
Closed by #2733
